fix compilation on top of zlib provided by Gentoo,
where OF macro has been renamed to _Z_OF
see https://bugreports.qt.io/browse/QTBUG-68467 and https://bugs.gentoo.org/383179
author: Kenneth Hoste (HPC-UGent)
diff -ru qt-everywhere-src-5.14.1.orig/qt3d/src/3rdparty/assimp/contrib/unzip/ioapi.h qt-everywhere-src-5.14.1/qt3d/src/3rdparty/assimp/contrib/unzip/ioapi.h
--- qt-everywhere-src-5.14.1.orig/qt3d/src/3rdparty/assimp/contrib/unzip/ioapi.h	2020-01-15 17:09:24.000000000 -0000
+++ qt-everywhere-src-5.14.1/qt3d/src/3rdparty/assimp/contrib/unzip/ioapi.h	2020-09-25 18:30:03.146291932 -0000
@@ -35,6 +35,15 @@
 extern "C" {
 #endif
 
+// if OF macro is not defined by _Z_OF is, define OF as _Z_OF
+// required when building with zlib provided by Gentoo
+// see https://bugreports.qt.io/browse/QTBUG-68467 and https://bugs.gentoo.org/383179
+#ifndef OF
+ #ifdef _Z_OF
+   #define OF _Z_OF
+ #endif
+#endif
+
 typedef voidpf (ZCALLBACK *open_file_func) OF((voidpf opaque, const char* filename, int mode));
 typedef uLong  (ZCALLBACK *read_file_func) OF((voidpf opaque, voidpf stream, void* buf, uLong size));
 typedef uLong  (ZCALLBACK *write_file_func) OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
diff -ru qt-everywhere-src-5.14.1.orig/qtquick3d/src/3rdparty/assimp/src/contrib/unzip/ioapi.h qt-everywhere-src-5.14.1/qtquick3d/src/3rdparty/assimp/src/contrib/unzip/ioapi.h
--- qt-everywhere-src-5.14.1.orig/qtquick3d/src/3rdparty/assimp/src/contrib/unzip/ioapi.h	2019-03-10 11:05:27.000000000 -0000
+++ qt-everywhere-src-5.14.1/qtquick3d/src/3rdparty/assimp/src/contrib/unzip/ioapi.h	2020-09-25 18:30:16.136640746 -0000
@@ -35,6 +35,15 @@
 extern "C" {
 #endif
 
+// if OF macro is not defined by _Z_OF is, define OF as _Z_OF
+// required when building with zlib provided by Gentoo
+// see https://bugreports.qt.io/browse/QTBUG-68467 and https://bugs.gentoo.org/383179
+#ifndef OF
+ #ifdef _Z_OF
+   #define OF _Z_OF
+ #endif
+#endif
+
 typedef voidpf (ZCALLBACK *open_file_func) OF((voidpf opaque, const char* filename, int mode));
 typedef uLong  (ZCALLBACK *read_file_func) OF((voidpf opaque, voidpf stream, void* buf, uLong size));
 typedef uLong  (ZCALLBACK *write_file_func) OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
diff -ru qt-everywhere-src-5.14.1.orig/qtquick3d/src/3rdparty/assimp/src/contrib/zlib/contrib/minizip/ioapi.h qt-everywhere-src-5.14.1/qtquick3d/src/3rdparty/assimp/src/contrib/zlib/contrib/minizip/ioapi.h
--- qt-everywhere-src-5.14.1.orig/qtquick3d/src/3rdparty/assimp/src/contrib/zlib/contrib/minizip/ioapi.h	2019-03-10 11:05:27.000000000 -0000
+++ qt-everywhere-src-5.14.1/qtquick3d/src/3rdparty/assimp/src/contrib/zlib/contrib/minizip/ioapi.h	2020-09-25 18:30:23.096828085 -0000
@@ -131,6 +131,15 @@
 
 
 
+// if OF macro is not defined by _Z_OF is, define OF as _Z_OF
+// required when building with zlib provided by Gentoo
+// see https://bugreports.qt.io/browse/QTBUG-68467 and https://bugs.gentoo.org/383179
+#ifndef OF
+ #ifdef _Z_OF
+   #define OF _Z_OF
+ #endif
+#endif
+
 
 typedef voidpf   (ZCALLBACK *open_file_func)      OF((voidpf opaque, const char* filename, int mode));
 typedef uLong    (ZCALLBACK *read_file_func)      OF((voidpf opaque, voidpf stream, void* buf, uLong size));
diff -ru qt-everywhere-src-5.14.1.orig/qtwebengine/src/3rdparty/chromium/third_party/zlib/contrib/minizip/ioapi.h qt-everywhere-src-5.14.1/qtwebengine/src/3rdparty/chromium/third_party/zlib/contrib/minizip/ioapi.h
--- qt-everywhere-src-5.14.1.orig/qtwebengine/src/3rdparty/chromium/third_party/zlib/contrib/minizip/ioapi.h	2020-01-20 10:37:42.000000000 -0000
+++ qt-everywhere-src-5.14.1/qtwebengine/src/3rdparty/chromium/third_party/zlib/contrib/minizip/ioapi.h	2020-09-25 18:29:45.635820801 -0000
@@ -129,8 +129,14 @@
  #endif
 #endif
 
-
-
+// if OF macro is not defined by _Z_OF is, define OF as _Z_OF
+// required when building with zlib provided by Gentoo
+// see https://bugreports.qt.io/browse/QTBUG-68467 and https://bugs.gentoo.org/383179
+#ifndef OF
+ #ifdef _Z_OF
+   #define OF _Z_OF
+ #endif
+#endif
 
 typedef voidpf   (ZCALLBACK *open_file_func)      OF((voidpf opaque, const char* filename, int mode));
 typedef uLong    (ZCALLBACK *read_file_func)      OF((voidpf opaque, voidpf stream, void* buf, uLong size));
